### PR TITLE
workflows: add Github action workflow of Golang for Windows

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu jackysp@gmail.com

### What problem does this PR solve? 
TiDB works well on Windows, but its tests sometimes are broken due to no CI guard them.

### What is changed and how it works?
Add a free Github action workflow for it.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by jackysp** [comment]
image
10 cases failed.

**Comment by tiancaiamao** [comment]
LGTM
Although there are not so many failed cases on my windows

File: ddl/db_partition_test.go [hunk 1]
```diff
@@ -1252,6 +1253,10 @@ func (s *testIntegrationSuite2) TestPartitionCancelAddPrimaryKey(c *C) {
 }
 
 func (s *testIntegrationSuite4) TestPartitionCancelAddIndex(c *C) {
+	if runtime.GOOS == "windows" {
+		// TODO: find the cause of the failure.
```
Comment (zz-jason) [inline ddl/db_partition_test.go]: It's better to create a github issue for it.

File: ddl/db_test.go [hunk 2]
```diff
@@ -355,6 +356,10 @@ LOOP:
 }
 
 func (s *testDBSuite5) TestCancelAddPrimaryKey(c *C) {
+	if runtime.GOOS == "windows" {
+		// TODO: find the cause of the failure.
```
Comment (zz-jason) [inline ddl/db_test.go]: ditto

File: ddl/db_partition_test.go (hunk 1 above)
Comment (jackysp) [reply ddl/db_partition_test.go]: No time to do this yet. I want to check these skipped tests after https://github.com/pingcap/tidb/projects/32.

**Comment by jackysp** [comment]
The CI performance provided by the GitHub action is too low. I will abandon this PR.

**Pull request closed without merging**